library/core/src/ptr/mod.rs

```diff
@@ -2285,6 +2285,13 @@ impl<F: FnPtr> fmt::Debug for F {
 /// `addr_of!(expr)` is equivalent to `&raw const expr`. The macro is *soft-deprecated*;
 /// use `&raw const` instead.
 ///
+/// It is still an open question whether writing through an `addr_of!`-created pointer is permitted
+/// or not. Specifically, if the place `expr` evaluates to is based on a raw pointer, then the
+/// result of `addr_of!` inherits all permissions from that raw pointer. However, if the place is
+/// based on a reference, local variable, or `static`, then until all details are decided, the same
+/// rules as for shared references apply: it is UB to write through a pointer created with this
```

Comment:
Saying it's UB feels a bit too dispositive given the situation.  Really, it's just undefined whether it's undefined, as it says above.

Reply:
Ah, nevermind, given #56604 it really is undefined.